mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -98,32 +100,55 @@ void AffixTuningParameters::runOnFunction() {
   func.walk([&](miopen::GridwiseGemmOp op) {
     OpBuilder b(op.getContext());
 
-    // TBD. Compute tuning parameters from actual logic.
-    op.setAttr("block_size", b.getI32IntegerAttr(256));
+    ConvolutionContext convContext = populateConvContext(op);
+    InitParamsNonXDL validParams{0, 0, 0, 0, 0, 0};
+    GemmSize gemmSize;
+    DerivedParams gemmADerivedParam;
+    DerivedParams gemmBDerivedParam;
+    int64_t gemmCDstPerWrite;
+    int64_t gridSize;
 
-    op.setAttr("m_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("n_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("k_per_block", b.getI32IntegerAttr(8));
+    PopulateParams populateParams;
+    populateParams.paramsFromCtx(convContext, validParams, gemmSize,
+                                 gemmADerivedParam, gemmBDerivedParam,
+                                 gemmCDstPerWrite, gridSize);
 
-    op.setAttr("m_per_thread", b.getI32IntegerAttr(4));
-    op.setAttr("n_per_thread", b.getI32IntegerAttr(4));
-    op.setAttr("k_per_thread", b.getI32IntegerAttr(1));
+    // Tunable parameters.
+    op.setAttr("block_size", b.getI32IntegerAttr(validParams.blockSize));
+    op.setAttr("m_per_block", b.getI32IntegerAttr(validParams.gemmMPerBlock));
+    op.setAttr("n_per_block", b.getI32IntegerAttr(validParams.gemmNPerBlock));
+    op.setAttr("k_per_block", b.getI32IntegerAttr(validParams.gemmKPerBlock));
+    op.setAttr("m_per_thread", b.getI32IntegerAttr(validParams.gemmMPerThread));
+    op.setAttr("n_per_thread", b.getI32IntegerAttr(validParams.gemmNPerThread));
+
+    // Derived parameters for gemmA.
+    op.setAttr("matrix_a_source_data_per_read",
+               b.getI32IntegerAttr(gemmADerivedParam.srcDataPerRead));
+    op.setAttr("matrix_a_dest_data_per_write_dim_m",
+               b.getI32IntegerAttr(gemmADerivedParam.dstDataPerWrite));
+
+    // Derived parameters for gemmB.
+    op.setAttr("matrix_b_source_data_per_read",
+               b.getI32IntegerAttr(gemmBDerivedParam.srcDataPerRead));
+    op.setAttr("matrix_b_dest_data_per_write_dim_n",
+               b.getI32IntegerAttr(gemmBDerivedParam.dstDataPerWrite));
+
+    // Derived parameters for gemmC.
+    // XXX. We only use 2D coordinates when storing VGPR to global VRAM.
```

Comment:
remove this XXX. We use 4D coordinates storing VGPR data to global VRAM for matrix C now.

Reply:
Will do.